ksql-engine/src/test/java/io/confluent/ksql/util/AvroUtilTest.java

```diff
@@ -0,0 +1,164 @@
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.ksql.metastore.KsqlTopic;
+import io.confluent.ksql.metastore.MetaStoreImpl;
+import io.confluent.ksql.parser.KsqlParser;
+import io.confluent.ksql.parser.tree.AbstractStreamCreateStatement;
+import io.confluent.ksql.parser.tree.Statement;
+import io.confluent.ksql.serde.DataSource;
+import io.confluent.ksql.serde.avro.KsqlAvroTopicSerDe;
+
+import static org.easymock.EasyMock.anyObject;
+import static org.easymock.EasyMock.anyString;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.mock;
+import static org.easymock.EasyMock.replay;
+import static org.junit.Assert.assertThat;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.fail;
+
+public class AvroUtilTest {
+
+  String ordersAveroSchemaStr = "{"
+                     + "\"namespace\": \"kql\","
+                     + " \"name\": \"orders\","
+                     + " \"type\": \"record\","
+                     + " \"fields\": ["
+                     + "     {\"name\": \"ordertime\", \"type\": \"long\"},"
+                     + "     {\"name\": \"orderid\",  \"type\": \"long\"},"
+                     + "     {\"name\": \"itemid\", \"type\": \"string\"},"
+                     + "     {\"name\": \"orderunits\", \"type\": \"double\"},"
+                     + "     {\"name\": \"arraycol\", \"type\": {\"type\": \"array\", \"items\": \"double\"}},"
+                     + "     {\"name\": \"mapcol\", \"type\": {\"type\": \"map\", \"values\": \"double\"}}"
+                     + " ]"
+                     + "}";
+
+  @Test
+  public void shouldPassAvroCheck() throws Exception {
+    SchemaRegistryClient schemaRegistryClient = mock(SchemaRegistryClient.class);
+    SchemaMetadata schemaMetadata = mock(SchemaMetadata.class);
+    expect(schemaMetadata.getSchema()).andReturn(ordersAveroSchemaStr);
+    expect(schemaMetadata.getId()).andReturn(1);
+    expect(schemaRegistryClient.getLatestSchemaMetadata(anyString())).andReturn(schemaMetadata);
+    replay(schemaMetadata);
+    replay(schemaRegistryClient);
+    AbstractStreamCreateStatement abstractStreamCreateStatement = getAbstractStreamCreateStatement
+        ("CREATE STREAM S1 WITH "
+                                                                   + "(kafka_topic='s1_topic', "
+                                     + "value_format='avro' );");
+    Pair<AbstractStreamCreateStatement, String> checkResult = AvroUtil.checkAndSetAvroSchema
+        (abstractStreamCreateStatement, new HashMap<>(), schemaRegistryClient);
+    AbstractStreamCreateStatement newAbstractStreamCreateStatement = checkResult.getLeft();
+    assertThat("Columns were not added correctly from Avro schema.",
+               newAbstractStreamCreateStatement.getElements().size(), equalTo(6));
+    assertThat("Invalid column type mapping. ", newAbstractStreamCreateStatement.getElements()
```

Comment:
Any reason why you don't just do:
`assertThat(newAbstractStreamCreateStatement.getElements(), equalTo(Arrays.asList(...))`

Reply:
Changed to checking the whole list.